### ISSUE TITLE
Limit PHP support for CMS 6 to PHP 8.3 only

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/.github#297]
As per our [major release policy](https://docs.silverstripe.org/en/5/project_governance/major_release_policy/#php-support-commitments):

> At launch, a Silverstripe CMS major release supports all PHP versions in full support. PHP versions in limited support are not supported at launch by new Silverstripe CMS major releases.

According to [the PHP supported versions page](https://www.php.net/supported-versions.php), PHP 8.1 already lost active support, and PHP 8.2 will lose active support on 31 December 2024 which is before CMS 6 will launch.

Therefore, of the versions of PHP currently stable, only PHP 8.3 is allowed at release (PHP 8.4 will release before then, but that will be handled under a separate card)

## Acceptance criteria
- Docs indicate CMS 6 supports PHP 8.3 only at launch
- CI doesn't try to run anything with PHP 8.1 or 8.2 for CMS 6

## PRs

- https://github.com/silverstripe/supported-modules/pull/40
- https://github.com/silverstripe/gha-ci/pull/148

The above must be merged first - some of the CI runs depend on it before they'll go green (notably config - I think the rest probably don't need it, from looking at them, but not 100% sure)

- https://github.com/silverstripe/recipe-kitchen-sink/pull/70
- silverstripe/silverstripe-contentreview#250
- silverstripe/silverstripe-testsession#92
- silverstripe/silverstripe-staticpublishqueue#199
- silverstripe/silverstripe-behat-extension#280
- silverstripe/silverstripe-admin#1811
- silverstripe/recipe-core#99
- silverstripe/silverstripe-iframe#96
- silverstripe/silverstripe-sharedraftcontent#252
- silverstripe/silverstripe-registry#104
- silverstripe/silverstripe-subsites#592
- silverstripe/silverstripe-asset-admin#1486
- silverstripe/silverstripe-spamprotection#125
- silverstripe/recipe-blog#57
- silverstripe/silverstripe-session-manager#211
- silverstripe/recipe-content-blocks#46
- silverstripe/silverstripe-securityreport#86
- silverstripe/silverstripe-mimevalidator#82
- silverstripe/silverstripe-restfulserver#132
- silverstripe/recipe-cms#90
- silverstripe/silverstripe-login-forms#196
- silverstripe/silverstripe-externallinks#140
- silverstripe/silverstripe-realme#155
- silverstripe/silverstripe-versionfeed#118
- silverstripe/recipe-form-building#39
- silverstripe/silverstripe-campaign-admin#322
- silverstripe/silverstripe-webauthn-authenticator#194
- silverstripe/silverstripe-elemental-fileblock#59
- silverstripe/silverstripe-config#121
- silverstripe/silverstripe-linkfield#320
- silverstripe/recipe-testing#25
- silverstripe/recipe-services#39
- silverstripe/silverstripe-siteconfig#174
- silverstripe/recipe-reporting-tools#48
- silverstripe/silverstripe-gridfieldqueuedexport#119
- silverstripe/recipe-authoring-tools#40
- silverstripe/silverstripe-framework#11345
- silverstripe/silverstripe-segment-field#121
- silverstripe/silverstripe-crontask#91
- silverstripe/silverstripe-documentconverter#93
- silverstripe/silverstripe-versioned#467
- silverstripe/silverstripe-installer#377
- silverstripe/silverstripe-textextraction#100
- silverstripe/silverstripe-elemental-bannerblock#159
- silverstripe/silverstripe-event-dispatcher#27
- silverstripe/silverstripe-auditor#89
- silverstripe/silverstripe-reports#192
- silverstripe/silverstripe-frameworktest#200
- silverstripe/silverstripe-environmentcheck#110
- silverstripe/silverstripe-userforms#1318
- silverstripe/silverstripe-sitewidecontent-report#102
- silverstripe/silverstripe-graphql#598
- silverstripe/silverstripe-lumberjack#166
- silverstripe/silverstripe-ldap#87
- silverstripe/silverstripe-taxonomy#121
- silverstripe/silverstripe-hybridsessions#111
- silverstripe/silverstripe-totp-authenticator#170
- silverstripe/silverstripe-versioned-admin#358
- silverstripe/silverstripe-assets#631
- silverstripe/silverstripe-mfa#560
- silverstripe/silverstripe-tagfield#302
- dnadesign/silverstripe-elemental-userforms#99
- silverstripe/silverstripe-elemental#1238
- silverstripe/silverstripe-cms#2990
- silverstripe/silverstripe-blog#777
- silverstripe/recipe-collaboration#36
- silverstripe/silverstripe-errorpage#119
- bringyourownideas/silverstripe-maintenance#237
- bringyourownideas/silverstripe-composer-update-checker#94
- symbiote/silverstripe-advancedworkflow#549
- symbiote/silverstripe-multivaluefield#116
- symbiote/silverstripe-gridfieldextensions#414
- symbiote/silverstripe-queuedjobs#443
- tractorcow-farm/silverstripe-fluent#874
- silverstripe/developer-docs#560
